File: 0516/withdraw-class.py
class Deposit():
    def __init__(self, init_balance):
        self.balance = init_balance

    def withdraw(self, amount):
        if amount > self.balance:
            print('Insufficient funds')
        self.balance = self.balance - amount
        return self.balance

kei = Deposit(100)
bill = Deposit(1000)

print(" kei(25) : {}".format(kei.withdraw(25)))
print("bill(25) : {}".format(bill.withdraw(25)))
print(" kei(25) : {}".format(kei.withdraw(25)))
print("bill(25) : {}".format(bill.withdraw(25)))
print(" kei(60) : {}".format(kei.withdraw(60)))
print("bill(60) : {}".format(bill.withdraw(60)))

File: 0516/withdraw_class.cpp
#include <iostream>

class Deposit {
  private:
  int balance;
  public:
  Deposit(int init_balance) {
    balance = init_balance;
  }

  int withdraw(int amount) {
    if ( amount > balance ) {
      std::cerr << "Insufficient funds" << std::endl;
    } else {
      balance = balance - amount;
    }
    return balance;
  }
};

int main() {
  Deposit kei = Deposit(100);
  Deposit bill = Deposit(1000);

  //std::cout << "kei  :" << kei.balance << std::endl;
  std::cout << "kei  :" << kei.withdraw(25) << std::endl;
  std::cout << "bill :" << bill.withdraw(25) << std::endl;

  std::cout << "kei  :" << kei.withdraw(25) << std::endl;
  std::cout << "bill :" << bill.withdraw(25) << std::endl;

  std::cout << "kei  :" << kei.withdraw(60) << std::endl;
  std::cout << "bill :" << bill.withdraw(60) << std::endl;
}
